UI/basicGUI/src/pages/home/home.ts

```diff
@@ -236,27 +241,45 @@ loadLegend(){
       return;
     }
     //ToDo: change url to deployed server version
-    var url = "http://localhost:8081/navigation/start/"+start+"/destination/"+destination+"/driving/Zihan"
-    const http = require('http')
-    http.get(url, (resp) => {
-      let data = '';
-
-      // A chunk of data has been recieved.
-      resp.on('data', (chunk) => {
-        data += chunk;
-      });
+    // var url = "http://localhost:8081/navigation/start/"+start+"/destination/"+destination+"/driving/Zihan"
+    // const http = require('http')
+    // http.get(url, (resp) => {
+    //   let data = '';
+    //
+    //   // A chunk of data has been recieved.
+    //   resp.on('data', (chunk) => {
+    //     data += chunk;
+    //   });
+    //
+    //   // The whole response has been received. Print out the result.
+    //   resp.on('end', () => {
+    //     this.drawPolyline(data, this.map, this.startMarker, this.destMarker, this.weatherMarker)
+    //   });
+    //
+    // }).on("error", (err) => {
+    //   console.log("Error: " + err.message);
+    // });
+    var requestCallback = function(token, me) {
+        console.log("Token: " + token)
+        var httpOptions = {
+            headers: new HttpHeaders({
+                // 'Content-Type': 'multipart/form-data;charset=UTF-8',
+                'Accept': 'application/json;charset=UTF-8',
+                'Authorization': 'Bearer ' + token,
+            })
+        }
 
-      // The whole response has been received. Print out the result.
-      resp.on('end', () => {
-        this.drawPolyline(data, this.map, this.startMarker, this.destMarker, this.weatherMarker)
-      });
+        var url = "http://localhost:8081/navigation/start/"+start+"/destination/"+destination+"/driving"
```

Comment:
Would want to change it to the IP variable name. @zihan0 can you co-ordinate this?